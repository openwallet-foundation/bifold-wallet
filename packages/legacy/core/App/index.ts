/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@credo-ts/core'
import AgentProvider from '@credo-ts/react-hooks'

import App from './App'
import * as components from './components'
import { Button as IButton, ButtonImpl as Button, ButtonType } from './components/buttons/Button'
import IconButton, { ButtonLocation } from './components/buttons/IconButton'
import BulletPoint from './components/inputs/BulletPoint'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import ContentGradient from './components/misc/ContentGradient'
import CredentialCard from './components/misc/CredentialCard'
import InfoBox, { InfoBoxType } from './components/misc/InfoBox'
import ErrorModal from './components/modals/ErrorModal'
import NetInfo from './components/network/NetInfo'
import Record from './components/record/Record'
import InfoTextBox from './components/texts/InfoTextBox'
import Link from './components/texts/Link'
import Text from './components/texts/Text'
import { ToastType } from './components/toast/BaseToast'
import toastConfig from './components/toast/ToastConfig'
import { AttachTourStep } from './components/tour/AttachTourStep'
import { credentialOfferTourSteps } from './components/tour/CredentialOfferTourSteps'
import { credentialsTourSteps } from './components/tour/CredentialsTourSteps'
import { homeTourSteps } from './components/tour/HomeTourSteps'
import { proofRequestTourSteps } from './components/tour/ProofRequestTourSteps'
import { TourBox } from './components/tour/TourBox'
import LimitedTextInput from './components/inputs/LimitedTextInput'
import HomeFooterView from './components/views/HomeFooterView'
import KeyboardView from './components/views/KeyboardView'
import NotificationListItem from './components/listItems/NotificationListItem'
import * as contexts from './contexts'
import { AutoLockTime, ActivityProvider, useActivity } from './contexts/activity'
import { AuthProvider } from './contexts/auth'
import { NetworkProvider } from './contexts/network'
import { useTour } from './contexts/tour/tour-context'
import { TourProvider } from './contexts/tour/tour-provider'
import AttemptLockout from './screens/AttemptLockout'
import Developer from './screens/Developer'
import OnboardingPages from './screens/OnboardingPages'
import Preface from './screens/Preface'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import UseBiometry from './screens/UseBiometry'
import { loadLoginAttempt } from './services/keychain'
import { BifoldLogger } from './services/logger'
import * as types from './types'
import Scan from './screens/Scan'
import Onboarding from './screens/Onboarding'
import { DefaultScreenOptionsDictionary, useDefaultStackOptions } from './navigators/defaultStackOptions'
import { PINRules, walletTimeout } from './constants'
import { CredentialListFooterProps } from './types/credential-list-footer'
import { OpenIDCredentialRecordProvider } from './modules/openid/context/OpenIDCredentialRecordProvider'
import { defaultConfig, defaultHistoryEventsLogger } from './container-impl'
import useInitializeAgent from './hooks/initialize-agent'
import { DefaultScreenLayoutOptions } from './navigators/defaultLayoutOptions'

export * from './navigators'
export * from './services/storage'
export * from './types/attestation'
export { LocalStorageKeys } from './constants'
export * from './services/storage'
export { initLanguages, initStoredLanguage, translationResources, Locales } from './localization'
export { defaultState, mergeReducers, StoreProvider, StoreContext, useStore } from './contexts/store'
export { default as Store, DispatchAction, reducer } from './contexts/reducers/store'
export { useDeepLinks } from './hooks/deep-links'
export { Assets as ImageAssets } from './theme'
export { ThemeProvider, useTheme } from './contexts/theme'
export { AnimatedComponentsProvider, useAnimatedComponents } from './contexts/animated-components'
export { ColorPallet } from './theme'
export { animatedComponents } from './animated-components'
export { theme } from './theme'
export { useAuth } from './contexts/auth'
export { useNetwork } from './contexts/network'
export { NavigationTheme } from './theme'
export { testIdWithKey, testIdForAccessabilityLabel } from './utils/testable'
export { Screens, Stacks, TabStacks } from './types/navigators'
export { createStyles } from './screens/OnboardingPages'
export { statusBarStyleForColor, StatusBarStyles } from './utils/luminance'
export { BifoldError } from './types/error'
export { EventTypes } from './constants'
export { migrateToAskar } from './utils/migration'
export { createLinkSecretIfRequired, getAgentModules } from './utils/agent'
export { removeExistingInvitationIfRequired, connectFromScanOrDeepLink, formatTime, useCredentialConnectionLabel, getConnectionName } from './utils/helpers'
export { isValidAnonCredsCredential, getCredentialIdentifiers } from './utils/credential'
export { buildFieldsFromAnonCredsCredential } from './utils/oca'

export type { AnimatedComponents } from './animated-components'
export type {
  ISVGAssets,
  IFontAttributes,
  IInputAttributes,
  IInputs,
  ITextTheme,
  IBrandColors,
  ISemanticColors,
  INotificationColors,
  IGrayscaleColors,
  IColorPallet,
  IAssets,
  ITheme,
} from './theme'
export type { PersistentState } from './types/state'
export type { BifoldAgent } from './utils/agent'
export type { TourStep, RenderProps } from './contexts/tour/tour-context'
export type { GenericFn } from './types/fn'
export type {
  AuthenticateStackParams,
  OnboardingStackParams,
  NotificationStackParams,
  ContactStackParams,
} from './types/navigators'
export type { OnboardingStyleSheet }
export type { WalletSecret } from './types/security'
export type { ReducerAction } from './contexts/reducers/store'
export type {
  State,
  Onboarding as OnboardingState,
  LoginAttempt as LoginAttemptState,
  Preferences as PreferencesState,
  Migration as MigrationState,
  Tours as ToursState,
} from './types/state'

export type { InlineMessageProps } from './components/inputs/InlineErrorText'

export { InlineErrorPosition } from './types/error'

export type { CredentialListFooterProps }
export * from './container-api'
export { MainContainer } from './container-impl'

export type { ScreenLayoutConfig } from './types/navigators'
export type { HistoryEventsLoggerConfig } from './types/config'

export {
  App,
  Agent,
  AgentProvider,
  AuthProvider,
  NetworkProvider,
  TourProvider,
  useTour,
  AttachTourStep,
  TourBox,
  defaultConfig,
  defaultHistoryEventsLogger,
  homeTourSteps,
  credentialsTourSteps,
  credentialOfferTourSteps,
  proofRequestTourSteps,
  ButtonType,
  ButtonLocation,
  CheckBoxRow,
  CredentialCard,
  ContentGradient,
  ErrorModal,
  IconButton,
  ActivityProvider,
  useActivity,
  InfoTextBox,
  InfoBox,
  InfoBoxType,
  Link,
  AutoLockTime,
  ToastType,
  toastConfig,
  NetInfo,
  OnboardingPages,
  OpenIDCredentialRecordProvider,
  NotificationListItem,
  useDefaultStackOptions,
  useInitializeAgent,
  Splash,
  Developer,
  Terms,
  Preface,
  HomeFooterView as HomeContentView,
  UseBiometry,
  AttemptLockout,
  Record,
  Scan,
  Onboarding,
  types,
  components,
  contexts,
  Text,
  loadLoginAttempt,
  Button,
  BifoldLogger,
  LimitedTextInput,
  KeyboardView,
  BulletPoint,
  PINRules,
  walletTimeout,
  DefaultScreenOptionsDictionary,
  DefaultScreenLayoutOptions,
}
export type { IButton }
