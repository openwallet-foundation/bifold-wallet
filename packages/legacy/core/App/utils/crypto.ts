import argon2 from 'react-native-argon2'

export const hashPIN = async (PIN: string, salt: string): Promise<string> => {
  try {
    const result = await argon2(PIN, salt, {})
    const { rawHash } = result

    return rawHash
  } catch (error) {
    throw new Error(`Error generating hash for PIN ${String((error as Error)?.message ?? error)}`)
  }
}
