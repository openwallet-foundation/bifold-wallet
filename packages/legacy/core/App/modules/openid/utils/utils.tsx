import { getDomainFromUrl } from '@credo-ts/core'
import { Attribute, Field } from '@hyperledger/aries-oca/build/legacy'
import { W3cCredentialDisplay } from '../types'

/**
 * Converts a camelCase string to a sentence format (first letter capitalized, rest in lower case).
 * i.e. sanitizeString("helloWorld")  // returns: 'Hello world'
 */
export function sanitizeString(str: string) {
  const result = str.replace(/([a-z0-9])([A-Z])/g, '$1 $2').replaceAll('_', ' ')
  let words = result.split(' ')
  words = words.map((word, index) => {
    if (index === 0 || word.toUpperCase() === word) {
      return word.charAt(0).toUpperCase() + word.slice(1)
    }
    return word.charAt(0).toLowerCase() + word.slice(1)
  })
  return words.join(' ')
}

export function getHostNameFromUrl(url: string) {
  try {
    return getDomainFromUrl(url)
  } catch (error) {
    throw new Error(`Error getting hostname from url: ${error}`)
  }
}

export const buildFieldsFromOpenIDTemplate = (data: { [key: string]: unknown }): Array<Field> => {
  const fields = []
  for (const key of Object.keys(data)) {
    // omit id and type
    if (key === 'id' || key === 'type') continue

    let pushedVal: string | number | null = null
    if (typeof data[key] === 'string' || typeof data[key] === 'number') {
      pushedVal = data[key] as string | number | null
    }
    fields.push(new Attribute({ name: key, value: pushedVal }))
  }
  return fields
}

export function formatDate(input: string | Date): string {
  const date = input instanceof Date ? input : new Date(input)
  return date.toLocaleString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    hour12: true,
  })
}

export const getAttributeField = (display: W3cCredentialDisplay, searchKey: string): Attribute | undefined => {
  for (const [key, value] of Object.entries(display.attributes)) {
    if (searchKey === key) {
      return new Attribute({
        name: key,
        value: value as string | number | null,
        mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
      })
    }
  }
  return undefined
}
