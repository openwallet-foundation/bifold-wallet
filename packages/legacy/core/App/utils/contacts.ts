import { BasicMessageRecord, ConnectionRecord, CredentialExchangeRecord, ProofExchangeRecord } from '@credo-ts/core'

import { BifoldAgent } from './agent'

interface ConnectionWithMessages {
  conn: ConnectionRecord
  msgs: (BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord)[]
}

interface ConnectionWithLatestMessage {
  conn: ConnectionRecord
  latestMsg: BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord
}

/**
 * Function to fetch contacts (connections) in order of latest chat message without using hooks
 * @param agent - Credo agent
 * @returns ConnectionRecord[] sorted by most recent message
 */
export const fetchContactsByLatestMessage = async (agent: BifoldAgent, connections: ConnectionRecord[]): Promise<ConnectionRecord[]> => {
  const connectionsWithMessages = await Promise.all<ConnectionWithMessages>(
    connections.map(
      async (conn: ConnectionRecord): Promise<ConnectionWithMessages> => ({
        conn,
        msgs: [
          ...(await agent.basicMessages.findAllByQuery({ connectionId: conn.id })),
          ...(await agent.proofs.findAllByQuery({ connectionId: conn.id })),
          ...(await agent.credentials.findAllByQuery({ connectionId: conn.id })),
        ],
      })
    )
  )

  const connectionsWithLatestMessage: ConnectionWithLatestMessage[] = connectionsWithMessages.map((pair) => {
    return {
      conn: pair.conn,
      latestMsg: pair.msgs.reduce(
        (acc, cur) => {
          const accDate = acc.updatedAt || acc.createdAt
          const curDate = cur.updatedAt || cur.createdAt
          return accDate > curDate ? acc : cur
        },
        // Initial value if no messages exist for this connection is a placeholder with the date the connection was created
        { createdAt: pair.conn.createdAt } as BasicMessageRecord | CredentialExchangeRecord | ProofExchangeRecord
      ),
    }
  })

  return connectionsWithLatestMessage
    .sort(
      (a, b) =>
        new Date(b.latestMsg.updatedAt || b.latestMsg.createdAt).valueOf() -
        new Date(a.latestMsg.updatedAt || a.latestMsg.createdAt).valueOf()
    )
    .map((pair) => pair.conn)
}
