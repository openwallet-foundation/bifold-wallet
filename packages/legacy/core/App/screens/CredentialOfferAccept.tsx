import { CredentialState } from '@credo-ts/core'
import { useCredentialById } from '@credo-ts/react-hooks'
import { useNavigation } from '@react-navigation/native'
import React, { useCallback, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { AccessibilityInfo, ScrollView, StyleSheet, Text, View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import Button, { ButtonType } from '../components/buttons/Button'
import SafeAreaModal from '../components/modals/SafeAreaModal'
import { useAnimatedComponents } from '../contexts/animated-components'
import { useTheme } from '../contexts/theme'
import { Screens, TabStacks } from '../types/navigators'
import { testIdWithKey } from '../utils/testable'
import { TOKENS, useServices } from '../container-api'

enum DeliveryStatus {
  Pending,
  Completed,
  Declined,
}

export interface CredentialOfferAcceptProps {
  visible: boolean
  credentialId: string
  confirmationOnly?: boolean
}

const CredentialOfferAccept: React.FC<CredentialOfferAcceptProps> = ({ visible, credentialId, confirmationOnly }) => {
  const { t } = useTranslation()
  const [shouldShowDelayMessage, setShouldShowDelayMessage] = useState<boolean>(false)
  const [credentialDeliveryStatus, setCredentialDeliveryStatus] = useState<DeliveryStatus>(DeliveryStatus.Pending)
  const [timerDidFire, setTimerDidFire] = useState<boolean>(false)
  const [timer, setTimer] = useState<NodeJS.Timeout>()
  const credential = useCredentialById(credentialId)
  const navigation = useNavigation()
  const { ListItems } = useTheme()
  const { CredentialAdded, CredentialPending } = useAnimatedComponents()
  const [{ connectionTimerDelay }] = useServices([TOKENS.CONFIG])
  const connTimerDelay = connectionTimerDelay ?? 10000 // in ms
  const styles = StyleSheet.create({
    container: {
      ...ListItems.credentialOfferBackground,
      height: '100%',
      padding: 20,
    },
    image: {
      marginTop: 20,
    },
    messageContainer: {
      alignItems: 'center',
    },
    messageText: {
      textAlign: 'center',
      marginTop: 30,
    },
    controlsContainer: {
      marginTop: 'auto',
      margin: 20,
    },
    delayMessageText: {
      textAlign: 'center',
      marginTop: 20,
    },
  })

  if (!credential && !confirmationOnly) {
    throw new Error('Unable to fetch credential from Credo')
  }

  const onBackToHomeTouched = useCallback(() => {
    navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
  }, [navigation])

  const onDoneTouched = useCallback(() => {
    navigation.getParent()?.navigate(TabStacks.CredentialStack, { screen: Screens.Credentials })
  }, [navigation])

  useEffect(() => {
    if (!credential) {
      return
    }
    if (credential.state === CredentialState.CredentialReceived || credential.state === CredentialState.Done) {
      timer && clearTimeout(timer)
      setCredentialDeliveryStatus(DeliveryStatus.Completed)
    }
  }, [credential, timer])

  useEffect(() => {
    if (confirmationOnly) {
      timer && clearTimeout(timer)
      setCredentialDeliveryStatus(DeliveryStatus.Completed)
    }
  }, [confirmationOnly, timer])

  useEffect(() => {
    if (timerDidFire || credentialDeliveryStatus !== DeliveryStatus.Pending || !visible) {
      return
    }

    const timer = setTimeout(() => {
      setShouldShowDelayMessage(true)
      setTimerDidFire(true)
    }, connTimerDelay)

    setTimer(timer)

    return () => {
      timer && clearTimeout(timer)
    }
  }, [timerDidFire, credentialDeliveryStatus, visible, connTimerDelay])

  useEffect(() => {
    if (shouldShowDelayMessage && credentialDeliveryStatus !== DeliveryStatus.Completed) {
      AccessibilityInfo.announceForAccessibility(t('Connection.TakingTooLong'))
    }
  }, [shouldShowDelayMessage, credentialDeliveryStatus, t])

  return (
    <SafeAreaModal visible={visible} transparent={true} animationType={'none'}>
      <SafeAreaView style={{ ...ListItems.credentialOfferBackground }}>
        <ScrollView style={styles.container}>
          <View style={styles.messageContainer}>
            {credentialDeliveryStatus === DeliveryStatus.Pending && (
              <Text
                style={[ListItems.credentialOfferTitle, styles.messageText]}
                testID={testIdWithKey('CredentialOnTheWay')}
              >
                {t('CredentialOffer.CredentialOnTheWay')}
              </Text>
            )}

            {credentialDeliveryStatus === DeliveryStatus.Completed && (
              <Text
                style={[ListItems.credentialOfferTitle, styles.messageText]}
                testID={testIdWithKey('CredentialAddedToYourWallet')}
              >
                {t('CredentialOffer.CredentialAddedToYourWallet')}
              </Text>
            )}
          </View>

          <View style={[styles.image, { minHeight: 250, alignItems: 'center', justifyContent: 'flex-end' }]}>
            {credentialDeliveryStatus === DeliveryStatus.Completed && <CredentialAdded />}
            {credentialDeliveryStatus === DeliveryStatus.Pending && <CredentialPending />}
          </View>

          {shouldShowDelayMessage && credentialDeliveryStatus === DeliveryStatus.Pending && (
            <Text
              style={[ListItems.credentialOfferDetails, styles.delayMessageText]}
              testID={testIdWithKey('TakingTooLong')}
            >
              {t('Connection.TakingTooLong')}
            </Text>
          )}
        </ScrollView>

        <View style={styles.controlsContainer}>
          {credentialDeliveryStatus === DeliveryStatus.Pending && (
            <View>
              <Button
                title={t('Loading.BackToHome')}
                accessibilityLabel={t('Loading.BackToHome')}
                testID={testIdWithKey('BackToHome')}
                onPress={onBackToHomeTouched}
                buttonType={ButtonType.ModalSecondary}
              />
            </View>
          )}

          {credentialDeliveryStatus === DeliveryStatus.Completed && (
            <View>
              <Button
                title={t('Global.Done')}
                accessibilityLabel={t('Global.Done')}
                testID={testIdWithKey('Done')}
                onPress={onDoneTouched}
                buttonType={ButtonType.ModalPrimary}
              />
            </View>
          )}
        </View>
      </SafeAreaView>
    </SafeAreaModal>
  )
}

export default CredentialOfferAccept
