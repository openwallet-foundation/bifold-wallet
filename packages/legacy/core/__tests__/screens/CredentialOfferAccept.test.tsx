import { CredentialExchangeRecord as CredentialRecord, CredentialState } from '@credo-ts/core'
import { useCredentialById } from '@credo-ts/react-hooks'
import { act, render } from '@testing-library/react-native'
import fs from 'fs'
import path from 'path'
import React from 'react'

import CredentialOfferAccept from '../../App/screens/CredentialOfferAccept'
import { testIdWithKey } from '../../App/utils/testable'
// import timeTravel from '../helpers/timetravel'
import { BasicAppContext } from '../helpers/app'

const credentialId = '123'
const credentialPath = path.join(__dirname, '../fixtures/degree-credential.json')
const credential = JSON.parse(fs.readFileSync(credentialPath, 'utf8'))
const credentialRecord = new CredentialRecord(credential)
// TODO:(jl) Make a fn to revive JSON dates properly and pass to `parse`
credentialRecord.createdAt = new Date(credentialRecord.createdAt)

// @ts-expect-error useCredentialById will be replaced with a mock which does have this method
useCredentialById.mockReturnValue(credentialRecord)

describe('CredentialOfferAccept Screen', () => {
  test('renders correctly', () => {
    const tree = render(
      <BasicAppContext>
        <CredentialOfferAccept visible={true} credentialId={credentialId} />
      </BasicAppContext>
    )

    const doneButton = tree.queryByTestId('Done')

    expect(tree).toMatchSnapshot()
    expect(doneButton).toBeNull()
    expect(useCredentialById).toBeCalledWith(credentialId)
  })

  test('transitions to taking too delay message', async () => {
    jest.useFakeTimers()
    const tree = render(
      <BasicAppContext>
        <CredentialOfferAccept visible={true} credentialId={credentialId} />
      </BasicAppContext>
    )

    await act(async () => {
      await jest.advanceTimersByTimeAsync(11000)
    })

    const backToHomeButton = tree.getByTestId(testIdWithKey('BackToHome'))
    const doneButton = tree.queryByTestId(testIdWithKey('Done'))

    expect(tree).toMatchSnapshot()
    expect(backToHomeButton).not.toBeNull()
    expect(doneButton).toBeNull()
  })

  test('transitions to offer accepted', () => {
    credentialRecord.state = CredentialState.CredentialReceived

    const tree = render(
      <BasicAppContext>
        <CredentialOfferAccept visible={true} credentialId={credentialId} />
      </BasicAppContext>
    )

    const doneButton = tree.getByTestId(testIdWithKey('Done'))
    const backToHomeButton = tree.queryByTestId(testIdWithKey('BackToHome'))

    expect(tree).toMatchSnapshot()
    expect(doneButton).not.toBeNull()
    expect(backToHomeButton).toBeNull()
  })
})
