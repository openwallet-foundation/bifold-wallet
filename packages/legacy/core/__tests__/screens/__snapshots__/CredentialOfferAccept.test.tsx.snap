// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CredentialOfferAccept Screen renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    edges={
      Object {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            maxFontSizeMultiplier={2}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 26,
                    "fontWeight": "normal",
                  },
                  Object {
                    "marginTop": 30,
                    "textAlign": "center",
                  },
                ],
              ]
            }
            testID="com.ariesbifold:id/CredentialOnTheWay"
          >
            CredentialOffer.CredentialOnTheWay
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 20,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "flex-end",
                "minHeight": 250,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <
              height={110}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "marginTop": -30,
                  "position": "absolute",
                }
              }
              width={110}
            />
            <View
              collapsable={false}
              style={
                Object {
                  "opacity": 0,
                  "transform": Array [
                    Object {
                      "translateY": -90,
                    },
                  ],
                }
              }
            >
              <
                height={110}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "marginLeft": 10,
                    "position": "absolute",
                  }
                }
                width={110}
              />
            </View>
            <
              height={140}
              style={
                Object {
                  "backgroundColor": "transparent",
                }
              }
              width={140}
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "margin": 20,
          "marginTop": "auto",
        }
      }
    >
      <View>
        <View
          accessibilityLabel="Loading.BackToHome"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#42803E",
              "borderRadius": 4,
              "borderWidth": 2,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/BackToHome"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Array [
                    Object {
                      "color": "#42803E",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                    false,
                    false,
                    false,
                  ],
                ]
              }
            >
              Loading.BackToHome
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`CredentialOfferAccept Screen transitions to offer accepted 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    edges={
      Object {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            maxFontSizeMultiplier={2}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 26,
                    "fontWeight": "normal",
                  },
                  Object {
                    "marginTop": 30,
                    "textAlign": "center",
                  },
                ],
              ]
            }
            testID="com.ariesbifold:id/CredentialAddedToYourWallet"
          >
            CredentialOffer.CredentialAddedToYourWallet
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 20,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "flex-end",
                "minHeight": 250,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "alignItems": "center",
                  "marginBottom": 10,
                  "opacity": 0,
                }
              }
            >
              <
                height={45}
                width={45}
              />
            </View>
            <View>
              <
                height={110}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "marginTop": -30,
                    "position": "absolute",
                  }
                }
                width={110}
              />
              <View
                collapsable={false}
                style={
                  Object {
                    "opacity": 0,
                    "transform": Array [
                      Object {
                        "translateY": -90,
                      },
                    ],
                  }
                }
              >
                <
                  height={110}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "marginLeft": 10,
                      "position": "absolute",
                    }
                  }
                  width={110}
                />
              </View>
              <
                height={140}
                style={
                  Object {
                    "backgroundColor": "transparent",
                  }
                }
                width={140}
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "margin": 20,
          "marginTop": "auto",
        }
      }
    >
      <View>
        <View
          accessibilityLabel="Global.Done"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#42803E",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/Done"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                    false,
                    false,
                    false,
                  ],
                ]
              }
            >
              Global.Done
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;

exports[`CredentialOfferAccept Screen transitions to taking too delay message 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <RNCSafeAreaView
    edges={
      Object {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "backgroundColor": "#000000",
          "height": "100%",
          "padding": 20,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            maxFontSizeMultiplier={2}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 26,
                    "fontWeight": "normal",
                  },
                  Object {
                    "marginTop": 30,
                    "textAlign": "center",
                  },
                ],
              ]
            }
            testID="com.ariesbifold:id/CredentialOnTheWay"
          >
            CredentialOffer.CredentialOnTheWay
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 20,
              },
              Object {
                "alignItems": "center",
                "justifyContent": "flex-end",
                "minHeight": 250,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "column",
              }
            }
          >
            <
              height={110}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "marginTop": -30,
                  "position": "absolute",
                }
              }
              width={110}
            />
            <View
              collapsable={false}
              style={
                Object {
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "translateY": -16.252128433746066,
                    },
                  ],
                }
              }
            >
              <
                height={110}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "marginLeft": 10,
                    "position": "absolute",
                  }
                }
                width={110}
              />
            </View>
            <
              height={140}
              style={
                Object {
                  "backgroundColor": "transparent",
                }
              }
              width={140}
            />
          </View>
        </View>
        <Text
          maxFontSizeMultiplier={2}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                },
                Object {
                  "marginTop": 20,
                  "textAlign": "center",
                },
              ],
            ]
          }
          testID="com.ariesbifold:id/TakingTooLong"
        >
          Connection.TakingTooLong
        </Text>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "margin": 20,
          "marginTop": "auto",
        }
      }
    >
      <View>
        <View
          accessibilityLabel="Loading.BackToHome"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            Object {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#42803E",
              "borderRadius": 4,
              "borderWidth": 2,
              "opacity": 1,
              "padding": 16,
            }
          }
          testID="com.ariesbifold:id/BackToHome"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              maxFontSizeMultiplier={2}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Array [
                    Object {
                      "color": "#42803E",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                    false,
                    false,
                    false,
                  ],
                ]
              }
            >
              Loading.BackToHome
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</Modal>
`;
